src/ws/ws_manager.rs

```diff
@@ -94,27 +93,53 @@ impl WsManager {
     pub(crate) async fn new(url: String) -> Result<WsManager> {
         let stop_flag = Arc::new(AtomicBool::new(false));
 
-        let (ws_stream, _) = connect_async(url.clone())
-            .await
-            .map_err(|e| Error::Websocket(e.to_string()))?;
-
-        let (writer, mut reader) = ws_stream.split();
+        let (writer, mut reader) = Self::connect(url.as_str()).await?.split();
```

Comment:
nit: prefer `&url` instead of `url.as_str` here and elsewhere

Reply:
Updated